### PR TITLE
AR-6350 mfa setup recovery for firebase within wallet

### PR DESCRIPTION
**Pull request by shrinathprabhu**
Resolves [AR-6350](https://team-1624093970686.atlassian.net/browse/AR-6350).

## Changes

- Open MFA setup and restore screen from within wallet for firebase

## Checklist

- [x] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [x] The changes have been tested locally.

[AR-6350]: https://team-1624093970686.atlassian.net/browse/AR-6350?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/MFARequiredScreen.vue [hunk 1]
```diff
@@ -24,56 +24,63 @@ let mfaWindow: Window | null
...
-    const handler = async (event: MessageEvent) => {
-      if (!event?.data?.status) {
-        return
-      }
-      cleanExit = true
-      const data = event.data
+  const { loginType } = JSON.parse(
+    storage.session.getItem('userInfo') as string
+  )
+  if (loginType === 'firebase') {
```
Comment (mmjee) [inline src/pages/MFARequiredScreen.vue:30]: It's better to use a Set membership check here, we are more likely to add more bearer type authentication methods.

File: src/pages/MFARestoreScreen.vue [hunk 2]
```diff
@@ -49,7 +53,18 @@ onBeforeMount(async () => {
     show: true,
     message: 'Loading metadata...',
   }
-  dkgShare = JSON.parse(storage.local.getItem('pk') as string)
+  const localSession = storage.session.getItem('userInfo')
+  if (localSession) {
+    userInfoSession = JSON.parse(localSession)
+  }
+  if (userInfoSession?.loginType === 'firebase') {
```
Comment (mmjee) [inline src/pages/MFARestoreScreen.vue:60]: Same, let's have a list of bearer type (non-social) authentication methods and use a set membership check.

File: src/pages/profileScreen.vue [hunk 3]
```diff
@@ -98,58 +99,66 @@ function handleShowMFAProceedModal(show: boolean) {
...
-  cleanExit = false
-  const mfaSetupPath = new URL(`mfa/${appStore.id}/setup`, AUTH_URL)
-  if (appStore.standaloneMode == 0) {
-    mfaWindow = window.open(
-      mfaSetupPath.toString(),
-      '_blank',
-      getWindowFeatures()
-    )
+  const { loginType } = JSON.parse(
+    getStorage().session.getItem('userInfo') as string
```
Comment (mmjee) [inline src/pages/profileScreen.vue:103]: It looks like we are relying on local storage to pass state around, don't we have a better way to do this?

File: src/pages/signInV2.vue [hunk 4]
```diff
@@ -139,9 +139,32 @@ async function fetchAvailableLogins(authProvider: AuthProvider) {
...
 
 async function storeUserInfoAndRedirect(
-  userInfo: GetInfoOutput & { hasMfa?: boolean; pk?: string }
+  userInfo: GetInfoOutput & {
+    hasMfa?: boolean
+    pk?: string
+  }
 ) {
   const storage = getStorage()
+  if ((userInfo.loginType as string) === 'firebase') {
```
Comment (mmjee) [inline src/pages/signInV2.vue:148]: I don't understand this, why are we initializing Core here? Why do we only do this for Firebase?

File: src/pages/MFARequiredScreen.vue (hunk 1 above)
Comment (shrinathprabhu) [reply src/pages/MFARequiredScreen.vue]: created a util that checks this based on a list [d256a90](https://github.com/arcana-network/wallet-ui/pull/353/commits/d256a909bd1547d303682852f8bf01298c7ad0ba)

File: src/utils/isInAppLogin.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+const inAppLoginTypes = ['firebase']
```
Comment (mmjee) [inline src/utils/isInAppLogin.ts:1]: Use a set here (`Set`), set membership checks are O(1) whereas array membership checks are O(n).

File: src/pages/signInV2.vue (hunk 4 above)
Comment (mmjee) [reply src/pages/signInV2.vue:150]: Can we handle this more generally? This is already pretty hacky.

File: src/pages/profileScreen.vue (hunk 3 above)
Comment (shrinathprabhu) [reply src/pages/profileScreen.vue:104]: I guess we can add this loginType to the store as well. But it wasn't fitting any store like app or user. Anyway we have these info in sessionStorage to persist login in a single tab, so I thought we could get this info from here. I'll add it to user store

File: src/utils/isInAppLogin.ts (hunk 5 above)
Comment (shrinathprabhu) [reply src/utils/isInAppLogin.ts:1]: Ok

Comment (shrinathprabhu) [reply src/utils/isInAppLogin.ts]: Done here [22b444e](https://github.com/arcana-network/wallet-ui/pull/353/commits/22b444e9441c62754ef3933702464d3707c3192d)

**Review by makylfang — APPROVED** [review]

**Pull request merged**